static/sidebar/js/libs/player.js

```diff
@@ -91,26 +83,34 @@
 
         //Context reference for Methods.
         var that = this;
+        this.getOption = function (name) {
+          if (this.options.hasOwnProperty(name))
+            return this.options.hasOwnProperty(name);
+          return null;
+        }
 
         //Sometimes we want/need to show an intearctive overlay on top of the player. We need this for MP4 videos that will
         //cue (mobile or autoplay:off) to actual play the video on demand.
-        this.addOverlay = function(imgUrl){
+        this.addOverlay = function(asset){
             var overlay = document.createElement('div');
-            overlay.classList.add('tvp-overlay');
-            overlay.style.backgroundImage = 'url("' + imgUrl + '")';
-            var overlayColor = this.overlayColor ? '#' + this.overlayColor : 'transparent';
-            overlay.innerHTML = '<div class="tvp-overlay-cover" style="opacity:' + this.overlayOpacity + ';background-image:linear-gradient(to bottom right,'+overlayColor+','+overlayColor+');"></div>'+
-                '<svg class="tvp-play" style="width:'+this.playButtonWidth+';height:'+this.playButtonHeight+';background-color:#'+this.playButtonBackgroundColor+';border:'+this.playButtonBorderWidth+' solid #'+this.playButtonBorderColor+';border-radius:'+this.playButtonBorderRadius+
-                '%;" viewBox="0 0 200 200"><polygon fill="#'+this.playButtonIconColor+'" points="70, 55 70, 145 145, 100"></polygon></svg>';
+            overlay.className = 'tvp-overlay';
+            overlay.style.backgroundImage = 'url("' + asset.thumbnailUrl + '")';
+            overlay.innerHTML = '<div class="tvp-overlay-cover"></div><svg class="tvp-play" viewBox="0 0 200 200">' +
```

Comment:
Let's move this HTML `<div class="tvp-overlay-cover">...` to `bootstrap.templates["modal-content"]`